### PR TITLE
allow for custom parameters when sending a screenView

### PR DESCRIPTION
**Pull request by sethladd**
Fixes https://github.com/dart-lang/usage/issues/112

Wasn't sure what you wanted me to do. But here's a start.

Also, not sure how to test this? I updated the test but it wasn't clear how to verify that it actually worked? The tests didn't fail.

I didn't find a grind task for running tests or analyzing, so I ran the tests by hand.

Thanks for taking a look!

### CODE REVIEW

File: lib/usage.dart [hunk 1]
```diff
@@ -236,8 +239,13 @@ class AnalyticsMock implements Analytics {
   @override
   String get clientId => '00000000-0000-4000-0000-000000000000';
 
-  Future sendScreenView(String viewName) =>
-      _log('screenView', {'viewName': viewName});
+  Future sendScreenView(String viewName, { Map<String, String> parameters }) {
+    if (parameters == null) {
```
Comment (devoncarew) [inline lib/usage.dart]: perhaps: `parameters ??= <String, String>{}`

File: test/usage_test.dart [hunk 2]
```diff
@@ -14,6 +14,7 @@ void defineTests() {
     test('simple', () {
       AnalyticsMock mock = new AnalyticsMock();
       mock.sendScreenView('main');
+      mock.sendScreenView('withParameters', parameters: {'cd1': 'custom'});
```
Comment (devoncarew) [inline test/usage_test.dart]: I would instead test this in test/usage_impl_test.dart, similar to how setSessionValue is tested.

**Comment by devoncarew** [comment]
> I didn't find a grind task for running tests or analyzing, so I ran the tests by hand.

Yeah, just pub run test is enough.

Thanks for the PR! Comments in-line.

**Comment by sethladd** [comment]
Want me to add this to sendEvent too?

Thanks for the review!

**Comment by devoncarew** [comment]
> Want me to add this to sendEvent too?

sgtm -

**Comment by sethladd** [comment]
Okey dokey!

**Comment by sethladd** [comment]
@devoncarew ptal thanks!

**Comment by devoncarew** [comment]
lgtm! thanks

**Comment by sethladd** [comment]
Thanks!

**Pull request merged**